2020/dec09/xmas.go

```diff
@@ -0,0 +1,36 @@
+package dec09
+
+import (
+	"fmt"
+)
+
+type XMAS struct {
+	preamble int
+}
+
+func (x XMAS) FindFirstInvalid(input []int) int {
+	i := x.preamble
+	// todo check that input length > preamble
+	// todo if everything is valid, you will explode
+	for ; x.IsValid(input, i); i++ {
+	}
+	return input[i]
+}
+
+func (x XMAS) IsValid(input []int, index int) bool {
+	_, _, err := findMatch2(input[index-x.preamble:index], input[index])
+	return err == nil       // no match found is the error
+}
+
+// findMatch2 is a reuse from December 1st
```

Comment:
It should therefore live in a lib. Or is it better to have a small copy than a large dependency?